flexmeasures/ui/utils/view_utils.py

```diff
@@ -22,6 +23,20 @@
 from flexmeasures.ui.utils.color_defaults import get_color_settings
 
 
+def fall_back_to_flask_template(render_function):
+    """In case the render_function is raising an error, fall back to using flask.render_template."""
+
+    @wraps(render_function)
+    def wrapper(template_name, *args, **kwargs):
+        try:
+            return render_function(template_name, *args, **kwargs)
+        except Exception:  # noqa: B902
+            return render_template(template_name, **kwargs)
```

Comment:
We should log the error here, otherwise it only gets swallowed.

Reply:
There are two errors and two viewpoints (server log and UI):

1. The original `ProgrammingError`

   > (we only show the original error in the HTML to the user)

   Actually, the original error did show up both in my server log and (with this PR) also in the UI.

2. An `psycopg2.errors.InFailedSqlTransaction` caused by rendering the flexmeasures template

   This came up in my server log before this PR, and you correctly pointed out it now got swallowed. With one of my latest revisions [49fef96](https://github.com/FlexMeasures/flexmeasures/pull/1302/commits/49fef9681634c2378e2c64c0b182fe6b56c4c5d4) the server log now shows it again. The UI still doesn't show this (and I don't think it needs to, nor that you requested this).

Reply:
Thanks!